test/util/util.go

```diff
@@ -1065,3 +1066,24 @@ func DeactivateWorkload(ctx context.Context, c client.Client, key client.ObjectK
 		g.Expect(c.Update(ctx, wl)).To(gomega.Succeed())
 	}, Timeout, Interval).Should(gomega.Succeed())
 }
+
+// conditionType corev1.NodeConditionType, conditionStatus corev1.ConditionStatus, time time.Time
+func SetNodeCondition(ctx context.Context, k8sClient client.Client, node *corev1.Node, newCondition *corev1.NodeCondition) {
+	gomega.EventuallyWithOffset(1, func(g gomega.Gomega) {
+		var updatedNode corev1.Node
+		g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(node), &updatedNode)).To(gomega.Succeed())
+		condition := utilnode.GetNodeCondition(&updatedNode, newCondition.Type)
+		changed := false
+		if condition == nil {
+			updatedNode.Status.Conditions = append(updatedNode.Status.Conditions, *newCondition)
+			changed = true
+		} else if condition.Status != newCondition.Status {
+			condition.Status = newCondition.Status
+			condition.LastTransitionTime = newCondition.LastTransitionTime
+			changed = true
+		}
+		if changed {
+			g.Expect(k8sClient.Status().Update(ctx, &updatedNode)).To(gomega.Succeed())
+		}
```

Comment:
nit, I would suggest to move the logic into utilnode.SetNodeCondition. This will allow us to unit test, and also it will be aligned with the approach for other conditions.